File: javascript/sieve/sieve.js
function Sieve(num) {
  this.primes = primeNums(num)
}

function primeNums(num) {
  var result = [2];
  for(var i = 3; i <= num; i++) {
    if (isPrime(i)) {
      result.push(i)
    }
  }
  return result;
}

function isPrime(value) {
  for(var i = 2; i < value; i++) {
    if(value % i === 0) {
        return false;
    }
  }
  return value > 1;
}

module.exports = Sieve;

File: python/sieve/sieve.py
def sieve(limit):
    result = []
    num = 0
    while num < limit:
        num += 1
        if is_prime(num):
            result.append(num)
    return result

def is_prime(num):
    if num <= 1:
        return False
    if num == 2:
        return True
    for n in range(3, num):
        if num % 2 == 0 or num % n == 0:
            return False
    return True
